chainstate/tx-verifier/src/transaction_verifier/accounting_delta_adapter.rs

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::BTreeMap;
+
+use super::{
+    error::ConnectTransactionError, storage::TransactionVerifierStorageError, TransactionSource,
+};
+use common::{
+    chain::{stakelock::StakePoolData, OutPoint, PoolId},
+    primitives::Amount,
+};
+use pos_accounting::{
+    DeltaMergeUndo, FlushablePoSAccountingView, PoSAccountingDelta, PoSAccountingDeltaData,
+    PoSAccountingOperations, PoSAccountingUndo, PoSAccountingView,
+};
+
+/// Adapter over `PosAccountingDelta` that mimics `PoSAccountingOperations`.
```

Comment:
Why not just implement that trait here?

Reply:
I need an additional argument `TransactionSource` for every function to accumulate deltas for a specific block.

Reply:
You can put it in the constructor of this struct

Reply:
Won't work. Nothing prevents the verifier to be used on multiple blocks, so I just pass it on every call

Reply:
You can still create an intermediary object that refers to the delta that holds the tx_source.

Meaning: PoSAccountingDeltaAdapter doesn't need to be owned by tx-verifier, but it can be created on demand with a reference to the delta object + contain the tx-source (which is light-weight). This way, you can still implement the trait to ensure that it respects the rules of that trait.

Reply:
Wanted this adapter to encapsulate the state as well so that there was no way to modify deltas except through this adapter. Which means that it has to hold the state. 
I could split this type in two: one that stores the state with a method that creates a lightweight accessor with references and implements the trait.

Reply:
Having something that implements the trait is a necessity to maintain consistency. Please give it a shot.